### PR TITLE
[feaLib][test] No need to create temporary file

### PR DESCRIPTION
**Pull request by khaledhosny**
Loading features from file objects have been supported for a while
already.

### CODE REVIEW

File: Lib/fontTools/feaLib/builder_test.py [hunk 1]
```diff
@@ -117,11 +118,8 @@ def expect_ttx(self, font, expected_ttx):
             self.fail("TTX output is different from expected")
 
     def build(self, featureFile):
-        path = self.temp_path(suffix=".fea")
-        with open(path, "w", encoding="utf-8") as outfile:
-            outfile.write(featureFile)
         font = makeTTFont()
-        addOpenTypeFeatures(font, path)
+        addOpenTypeFeaturesFromString(font, featureFile)
```
Comment (anthrotype) [inline Lib/fontTools/feaLib/builder_test.py]: Perhaps it might be worth having a separate test for parsing/building from a filesystem path input?

**Comment by anthrotype** [comment]
As you can see, the STAT tests are failing since @behdad added that extra field to the table header.
It's sill not clear to me whether we should revert the change or adjust the expected test results...
/cc @brawer @moyogo

File: Lib/fontTools/feaLib/parser_test.py [hunk 2]
```diff
@@ -1263,22 +1260,9 @@ def test_empty_statement_ignored(self):
...
-
     def parse(self, text):
-        if not self.tempdir:
-            self.tempdir = tempfile.mkdtemp()
-        self.num_tempfiles += 1
-        path = os.path.join(self.tempdir, "tmp%d.fea" % self.num_tempfiles)
-        with open(path, "w", encoding="utf-8") as outfile:
-            outfile.write(text)
-        return Parser(path).parse()
+        featurefile = UnicodeIO(tounicode(text))
```
Comment (anthrotype) [inline Lib/fontTools/feaLib/parser_test.py]: We are importing unicode_literals from __future__, so I think tounicode() should not be needed here.

File: Lib/fontTools/feaLib/builder_test.py (hunk 1 above)
Comment (khaledhosny) [inline Lib/fontTools/feaLib/builder_test.py]: If I’m reading the code correctly, there is already one few lines below.

Comment (anthrotype) [inline Lib/fontTools/feaLib/builder_test.py]: Oh ok! I was reading from my phone :)

File: Lib/fontTools/feaLib/parser_test.py (hunk 2 above)
Comment (khaledhosny) [inline Lib/fontTools/feaLib/parser_test.py]: Right.

**Pull request merged**